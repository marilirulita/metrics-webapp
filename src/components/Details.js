import { useLocation } from 'react-router-dom';
import { useEffect, useState } from 'react';
import { BsGraphUp } from 'react-icons/bs';
import { RiArrowRightCircleLine } from 'react-icons/ri';

const Details = () => {
  const location = useLocation();
  const [data, setData] = useState({});

  useEffect(() => {
    setData(location.state);
  }, [location]);

  const selectedData = ['sector', 'industry', 'country', 'marketCap', 'volume', 'lastAnnualDividend', 'beta'];

  const filteredData = selectedData.reduce((obj, key) => ({ ...obj, [key]: data[key] }), {});

  return (
    <div className="home-container">
      <div className="header-style">
        <span>
          <BsGraphUp color="#b13967" />
        </span>
        <div className="header-text-style">
          <h2>{data.companyName}</h2>
          <span>
            $
            {data.price}
          </span>
        </div>
      </div>
      <h4>COMPANY STOCK BREAKDOWN</h4>
      <div className="detail-section">
        { Object.keys(filteredData).map((key) => (
          <div key={key} className="details-data">
            <h3>
              {key}
              :
              {' '}
            </h3>
            <div className="details-text">
              <span>{filteredData[key]}</span>
              <span><RiArrowRightCircleLine size="30px" /></span>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};
export default Details;
