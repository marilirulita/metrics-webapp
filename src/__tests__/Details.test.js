import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Details from '../components/Details';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Provider store={store}>
          <Details />
        </Provider>
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
