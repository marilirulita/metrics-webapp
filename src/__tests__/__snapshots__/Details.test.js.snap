// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="home-container"
>
  <div
    className="header-style"
  >
    <span>
      <svg
        color="#b13967"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#b13967",
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h1v15h15v1H0V0zm10 3.5a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-1 0V4.9l-3.613 4.417a.5.5 0 0 1-.74.037L7.06 6.767l-3.656 5.027a.5.5 0 0 1-.808-.588l4-5.5a.5.5 0 0 1 .758-.06l2.609 2.61L13.445 4H10.5a.5.5 0 0 1-.5-.5z"
          fillRule="evenodd"
        />
      </svg>
    </span>
    <div
      className="header-text-style"
    >
      <h2 />
      <span>
        $
      </span>
    </div>
  </div>
  <h4>
    COMPANY STOCK BREAKDOWN
  </h4>
  <div
    className="detail-section"
  >
    <div
      className="details-data"
    >
      <h3>
        sector
        :
         
      </h3>
      <div
        className="details-text"
      >
        <span />
        <span>
          <svg
            fill="currentColor"
            height="30px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 11V8l4 4-4 4v-3H8v-2h4zm0-9c5.52 0 10 4.48 10 10s-4.48 10-10 10S2 17.52 2 12 6.48 2 12 2zm0 18c4.42 0 8-3.58 8-8s-3.58-8-8-8-8 3.58-8 8 3.58 8 8 8z"
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
    <div
      className="details-data"
    >
      <h3>
        industry
        :
         
      </h3>
      <div
        className="details-text"
      >
        <span />
        <span>
          <svg
            fill="currentColor"
            height="30px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 11V8l4 4-4 4v-3H8v-2h4zm0-9c5.52 0 10 4.48 10 10s-4.48 10-10 10S2 17.52 2 12 6.48 2 12 2zm0 18c4.42 0 8-3.58 8-8s-3.58-8-8-8-8 3.58-8 8 3.58 8 8 8z"
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
    <div
      className="details-data"
    >
      <h3>
        country
        :
         
      </h3>
      <div
        className="details-text"
      >
        <span />
        <span>
          <svg
            fill="currentColor"
            height="30px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 11V8l4 4-4 4v-3H8v-2h4zm0-9c5.52 0 10 4.48 10 10s-4.48 10-10 10S2 17.52 2 12 6.48 2 12 2zm0 18c4.42 0 8-3.58 8-8s-3.58-8-8-8-8 3.58-8 8 3.58 8 8 8z"
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
    <div
      className="details-data"
    >
      <h3>
        marketCap
        :
         
      </h3>
      <div
        className="details-text"
      >
        <span />
        <span>
          <svg
            fill="currentColor"
            height="30px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 11V8l4 4-4 4v-3H8v-2h4zm0-9c5.52 0 10 4.48 10 10s-4.48 10-10 10S2 17.52 2 12 6.48 2 12 2zm0 18c4.42 0 8-3.58 8-8s-3.58-8-8-8-8 3.58-8 8 3.58 8 8 8z"
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
    <div
      className="details-data"
    >
      <h3>
        volume
        :
         
      </h3>
      <div
        className="details-text"
      >
        <span />
        <span>
          <svg
            fill="currentColor"
            height="30px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 11V8l4 4-4 4v-3H8v-2h4zm0-9c5.52 0 10 4.48 10 10s-4.48 10-10 10S2 17.52 2 12 6.48 2 12 2zm0 18c4.42 0 8-3.58 8-8s-3.58-8-8-8-8 3.58-8 8 3.58 8 8 8z"
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
    <div
      className="details-data"
    >
      <h3>
        lastAnnualDividend
        :
         
      </h3>
      <div
        className="details-text"
      >
        <span />
        <span>
          <svg
            fill="currentColor"
            height="30px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 11V8l4 4-4 4v-3H8v-2h4zm0-9c5.52 0 10 4.48 10 10s-4.48 10-10 10S2 17.52 2 12 6.48 2 12 2zm0 18c4.42 0 8-3.58 8-8s-3.58-8-8-8-8 3.58-8 8 3.58 8 8 8z"
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
    <div
      className="details-data"
    >
      <h3>
        beta
        :
         
      </h3>
      <div
        className="details-text"
      >
        <span />
        <span>
          <svg
            fill="currentColor"
            height="30px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 11V8l4 4-4 4v-3H8v-2h4zm0-9c5.52 0 10 4.48 10 10s-4.48 10-10 10S2 17.52 2 12 6.48 2 12 2zm0 18c4.42 0 8-3.58 8-8s-3.58-8-8-8-8 3.58-8 8 3.58 8 8 8z"
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
